lightning/src/ln/channelmanager.rs

```diff
@@ -454,6 +454,17 @@ pub(crate) enum PendingOutboundPayment {
 		session_privs: HashSet<[u8; 32]>,
 		payment_hash: Option<PaymentHash>,
 	},
+	/// When a payer gives up trying to retry a payment, they inform us, letting us generate a
+	/// `PaymentFailed` event when all HTLCs have irrevocably failed. This avoids a number of race
+	/// conditions in MPP-aware payment retriers (1), where the possibility of multiple
+	/// `PaymentPathFailed` events with `all_paths_failed` can be pending at once, confusing a
+	/// downstream event handler as to when a payment has actually failed.
+	///
+	/// (1) https://github.com/lightningdevkit/rust-lightning/issues/1164
+	RetriesExceeded {
```

Comment:
Hmmmm, definitely think "abandon" is better than focusing on retries here, but I do worry that users will read that as "after I call abandon_payment, it will not succeed" which isn't true. Is there something we can come up with that doesn't imply the payment fails upon calling the method?

Reply:
I had some other ideas like "abort" and "discontinue", but "abandon" seemed a better fit, albeit not perfect. Compared to "abort", "abandon" doesn't seem as definitive on the ultimate outcome. I think the edge case that you mentioned could just be called out in the docs.

Reply:
Can we get away with `discontinue_retries`? It seems that focuses more on the action of calling `retry_payment` as the thing that's been "discontinued" vs some arbitrary in-another-place "retry counter"? If you disagree I'll go with abort.

Reply:
I'd prefer "abandon" over "abort" (just to clarify), but I could live with "discontinue". My preference for "abandon" was mostly to make the interface methods looks related, i.e.,`{send|retry|abandon}_payment`.

Reply:
Ah, yea, I hadn't really though of that symmetry. Going with abandon.